x-pack/plugins/ml/public/jobs/new_job/advanced/new_job_controller.js

```diff
@@ -1026,6 +1026,12 @@ module.controller('MlNewJob',
           tabs[0].checks.groupIds.message = msg;
         }
 
+        if (validationResults.contains('model_memory_limit_units_invalid')) {
+          tabs[0].checks.modelMemoryLimit.valid = false;
+          const msg = `Model memory limit data unit unrecognized. It must be B, KB, MB, GB, TB or PB`;
```

Comment:
Ideally the units in the message should come from the new `ALLOWED_DATA_UNITS ` constant